File: CLASE VII - Listas Void con Struct Anidadas/pre_parcial/golang/pasajero.go
package main

import "fmt"

type Pasajero struct {
	apellido   string
	dni        int
	ventanilla rune
}

func crearPasajero(apellido string, dni int, ventanilla rune) *Pasajero {
	return &Pasajero{apellido: apellido, dni: dni, ventanilla: ventanilla}
}

func (p *Pasajero) GetApellido() string  { return p.apellido }
func (p *Pasajero) GetDni() int          { return p.dni }
func (p *Pasajero) GetVentanilla() rune  { return p.ventanilla }
func (p *Pasajero) SetApellido(a string) { p.apellido = a }
func (p *Pasajero) SetDni(d int)         { p.dni = d }
func (p *Pasajero) SetVentanilla(v rune) { p.ventanilla = v }

func mostrarPasajero(p *Pasajero) {
	fmt.Printf("\n---------PASAJERO---------\n")
	fmt.Printf("\nAPELLIDO: %s\n", p.apellido)
	fmt.Printf("\nDNI: %d\n", p.dni)
	fmt.Printf("\nVENTANILLA: %c\n", p.ventanilla)
}

func mostrarPsajeroDatoPtr(p *Pasajero) {
	mostrarPasajero(p)
}

File: CLASE VII - Listas Void con Struct Anidadas/pre_parcial/c/pasajero.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "pasajero.h"

struct Pasajero{
    char apellido[20];
    int dni;
    char ventanilla;
};

PasajeroPtr crearPasajero(char* apellido, int dni, char ventanilla){
    PasajeroPtr p = (PasajeroPtr) malloc(sizeof(struct Pasajero));

    strcpy(p->apellido, apellido);
    p->dni = dni;
    p->ventanilla = ventanilla;

    return p;
};

char* getApellido(PasajeroPtr p){
    return p->apellido;
}
int getDni(PasajeroPtr p){
    return p->dni;
}
char getVentanilla(PasajeroPtr p){
    return p->ventanilla;
}

void setApellido(PasajeroPtr p, char* apellido){
    strcpy(p->apellido, apellido);
}
void setDni(PasajeroPtr p, int dni){
    p->dni = dni;
}
void setVentanilla(PasajeroPtr p, char ventanilla){
    p->ventanilla = ventanilla;
}

void mostrarPasajero(PasajeroPtr p){

    printf("\n---------PASAJERO---------\n");
    printf("\nAPELLIDO: %s\n", p->apellido);
    printf("\nDNI: %d\n", p->dni);
    printf("\nVENTANILLA: %c\n", p->ventanilla);
}


void mostrarPsajeroDatoPtr (void * p){

    PasajeroPtr pasajero = (PasajeroPtr )p;

    mostrarPasajero(pasajero);


}





void liberarPasajero(PasajeroPtr p){
    free(p);
}
